machine/machine_core/timeout.py

```diff
@@ -32,20 +38,21 @@ def __init__(self, seconds=1, error_message='Timeout', machine=None):
         self.error_message = error_message
         self.machine = machine
 
-    def handle_timeout(self, signum, frame):
+    def handle_timeout(self, signum: int, frame: Optional[FrameType]) -> None:
         if self.machine:
             if self.machine.ssh_process:
                 self.machine.ssh_process.terminate()
             self.machine.disconnect()
 
         raise RuntimeError(self.error_message)
 
-    def __enter__(self):
+    def __enter__(self) -> None:
         if self.seconds:
             signal.signal(signal.SIGALRM, self.handle_timeout)
             signal.alarm(self.seconds)
 
-    def __exit__(self, _type, value, traceback):
+    def __exit__(self, _type: Optional[Type[BaseException]], _value: Optional[BaseException],
```

Comment:
This is pretty frustrating...

If we don't care about these, I've seen it done like:

```python
def __exit__(self, *_args: object) -> None:
    ...
```